sdk/communication/Azure.Communication.Administration/src/Models/PhoneNumberReservationPurchaseOperation.cs

```diff
@@ -0,0 +1,161 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Administration.Models
+{
+    /// <summary>
+    /// Represents a long-running phone number reservation purchase operation.
+    /// </summary>
+    public class PhoneNumberReservationPurchaseOperation : Operation<ReservationStatus>
+    {
+        private bool _hasCompleted;
+        private readonly object _lockObject = new object();
+        private readonly PhoneNumberAdministrationClient _client;
+        private readonly CancellationToken _cancellationToken;
+        private ReservationStatus? _value;
+        private Response _rawResponse;
+        private Response _finalRawResponse;
+        private readonly IReadOnlyList<ReservationStatus> _terminateStatuses;
+
+        /// <summary>
+        /// Initializes a new <see cref="PhoneNumberReservationPurchaseOperation"/> instance.
+        /// </summary>
+        /// <param name="client">The client used to check for completion.</param>
+        /// <param name="id">The reservation operation ID.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public PhoneNumberReservationPurchaseOperation(
+            PhoneNumberAdministrationClient client,
+            string id,
+            CancellationToken cancellationToken = default)
+            : this(client, id, null!, cancellationToken)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="PhoneNumberReservationPurchaseOperation"/> instance
+        /// </summary>
+        /// <param name="client">The client used to check for completion.</param>
+        /// <param name="id">The reservation operation ID.</param>
+        /// <param name="initialResponse">The original server response on start operation request.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        internal PhoneNumberReservationPurchaseOperation(
+            PhoneNumberAdministrationClient client,
+            string id,
+            Response initialResponse,
+            CancellationToken cancellationToken = default)
+        {
+            _terminateStatuses = new[]
+            {
+                ReservationStatus.Success,
+                ReservationStatus.Expired,
+                ReservationStatus.Cancelled,
+                ReservationStatus.Error
+            };
+            Id = id;
+            _value = null;
+            _rawResponse = initialResponse;
+            _finalRawResponse = null!;
+            _client = client;
+            _cancellationToken = cancellationToken;
+        }
+
+        /// <inheritdocs />
+        public override string Id { get; }
+
+        /// <inheritdocs />
+        public override ReservationStatus Value => OperationHelpers.GetValue(ref _value);
```

Comment:
I know you personally don't like throwing here, but let's throw if the status isn't Success to be consistent with the other LROs.